### PR TITLE
seabios.bin_file: Add new case for seabios

### PR DESCRIPTION
**Pull request by ybduan**
Add bin file check for seabios

ID: 1482399

Signed-off-by: ybduan <yduan@redhat.com>

### CODE REVIEW

**Comment by ybduan** [comment]
(1/1) Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.seabios.bin_file: PASS (447.44 s)

**Comment by xiangchunfu** [comment]
@ybduan missed seabios.py file.

**Comment by ybduan** [comment]
(1/1) Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.seabios.bin_file: PASS (447.70 s)

File: qemu/tests/seabios_bin.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    1) Get available bin files
+    2) Get supported machine types
+    3) Check bin file in use
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    bin_dict = {'rhel6': 'bios.bin', 'rhel7': 'bios-256k.bin'}
```
Comment (xiywang) [inline qemu/tests/seabios_bin.py]: how about rhel8? same as rhel7? any need to add a pair of key and value?

Comment (ybduan) [reply qemu/tests/seabios_bin.py]: The supported machine type in rhel8 is not concluded, so no need to consider rhel8 for now. Thanks!

**Comment by ybduan** [comment]
Updated, please help review again, thanks! @xiangchunfu

File: qemu/tests/seabios_bin.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    for machine in machine_types:
+        error_context.context("Check bin file with machine type: %s" % machine,
+                              logging.info)
+        for key in bin_dict:
+            if key in machine:
+                bin_file = bin_dict[key]
+                break
+        else:
+            raise NotImplementedError()
```
Comment (vivianQizhu) [inline qemu/tests/seabios_bin.py]: You mean `test.cancel()` here?

Comment (ybduan) [reply qemu/tests/seabios_bin.py]: No, this part will also be updated when the supported machine types of rhel8 are concluded.

**Comment by ybduan** [comment]
Updated.

**Comment by ybduan** [comment]
(1/1) Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.seabios.bin_file: PASS (447.46 s)

File: qemu/tests/seabios_bin.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        machine_types.remove(i)
+    logging.info(machine_types)
+
+    for machine in machine_types:
+        error_context.context("Check bin file with machine type: %s" % machine,
+                              logging.info)
+        for key in bin_dict:
+            if key in machine:
+                bin_file = bin_dict[key]
+                break
```
[the remark this answers is not in the document]
Comment (xiangchunfu) [reply qemu/tests/seabios_bin.py]: @ybduan ignore this comment.  go through this code again.  this logic is correct.

**Comment by xiangchunfu** [comment]
Acked-by:Xiangchun Fu xfu@redhat.com

File: qemu/tests/seabios_bin.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    bin_dict = {'rhel6': 'bios.bin', 'rhel7': 'bios-256k.bin'}
+
+    error_context.context("Get available bin files", logging.info)
+    output = process.system_output('ls /usr/share/seabios', shell=True)
```
Comment (vivianQizhu) [inline qemu/tests/seabios_bin.py]: Should need decode for python3

File: qemu/tests/seabios_bin.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    for value in bin_dict.values():
+        if value not in output:
+            test.fail("%s is not available" % value)
+
+    error_context.context("Get supported machine types", logging.info)
+    qemu_binary = utils_misc.get_qemu_binary(params)
+    machine_type_cmd = qemu_binary + " -machine help | awk '{ print $1 }'"
+    output = process.system_output(machine_type_cmd, shell=True)
```
Comment (vivianQizhu) [inline qemu/tests/seabios_bin.py]: Same as above.

**Comment by ybduan** [comment]
(1/1) Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.seabios.bin_file: PASS (451.94 s)

 (1/1) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.io-github-autotest-qemu.seabios.bin_file: PASS (297.82 s)

File: qemu/tests/seabios_bin.py (hunk 4 above)
Comment (ybduan) [reply qemu/tests/seabios_bin.py]: Updated!

File: qemu/tests/seabios_bin.py (hunk 5 above)
Comment (ybduan) [reply qemu/tests/seabios_bin.py]: Updated!

**Pull request merged**